projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-flat/json-editor-node-flat/json-editor-node-flat.component.ts

```diff
@@ -53,19 +53,19 @@ export class JsonEditorNodeFlatComponent extends JsonEditorNode implements OnIni
 
   requiredIndicator: SafeHtml;
 
-  constructor(public dialogMngr: DialogService, private domSanitizer: DomSanitizer) {
-    super(dialogMngr);
-    this.requiredIndicator = this.domSanitizer.bypassSecurityTrustHtml(requiredIndicatorIcon);
-  }
-
-  ngOnInit() {
+  get nextLevel() {
```

Comment:
We only need to do this if/else once. No big deal in this case, but a getter is gonna keep doing this all the time.

Reply:
No sure about naming this `nextLevel`. It's really the actual level. If something could be changed is the @Input from `level` to `previousLevel`.

Reply:
I can remove the getter and mode the logic to on-changes.  Eisting code did `this.level += 1` which alterend the input and made the result not reactive to changes in input.

Reply:
Ok, it makes it we want to hide the root dynamically. I'd prefer `onChanges` but it's up to you.